packages/toggle-core/src/Read/ChainToggleStrategyFactory.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pheature\Core\Toggle\Read;
+
+use Pheature\Core\Toggle\Exception\InvalidStrategyTypeGiven;
+
+final class ChainToggleStrategyFactory
+{
+    /** @var ToggleStrategyFactory[] */
+    private array $toggleStrategyFactories;
+    private SegmentFactory $segmentFactory;
+
+    public function __construct(SegmentFactory $segmentFactory, ToggleStrategyFactory ...$toggleStrategyFactories)
+    {
+        $this->toggleStrategyFactories = $toggleStrategyFactories;
+        $this->segmentFactory = $segmentFactory;
```

Comment:
We could change the order of the assignments to follow the same order as the constructor arguments. Also for the class properties
```suggestion
        $this->segmentFactory = $segmentFactory;
$this->toggleStrategyFactories = $toggleStrategyFactories;
```